### PR TITLE
Add 18ZOO to game list

### PR DESCRIPTION
**Pull request by carlorusso**

### CODE REVIEW

**Comment by carlorusso** [comment]
Following https://github.com/tobymao/18xx/pull/3566
Contains all maps, configuration for draft step, end game configuration.

**Comment by tobymao** [comment]
excellent work, much better, will look at it soon thank you for putting in the effort to do this

File: lib/engine/game/g_18_zoo.rb [hunk 1]
```diff
@@ -0,0 +1,668 @@
...
+
+      def setup
+        @operating_rounds = 2 # 2 ORs on first and second round
+
+        @available_companies = []
+        @future_companies = []
+
+        draw_size = players.size == 5 ? 6 : 4
+        @companies_for_isr = @companies.first(draw_size)
+        @companies_for_monday = @companies.drop(draw_size + 4).first(4)
```
Comment (tobymao) [inline lib/engine/game/g_18_zoo.rb:505]: are you trying to slide here?

@companies[draw_size..draw_size + 4)

File: lib/engine/game/g_18_zoo.rb [hunk 2]
```diff
@@ -0,0 +1,668 @@
...
+      def new_stock_round
+        result = super
+
+        add_cousins if @turn == 3
+
+        update_current_and_future(@companies_for_monday, @companies_for_tuesday, 1)
+        update_current_and_future(@companies_for_tuesday, @companies_for_wednesday, 2)
+        update_current_and_future(@companies_for_wednesday, nil, 3)
+
+        @available_companies.select { |c| c.owner.nil? }.each { |c| c.owner = @bank }
```
Comment (tobymao) [inline lib/engine/game/g_18_zoo.rb:604]: @available_companies.each { |c| c.owner = @bank unless c.owner }

File: lib/engine/game/g_18_zoo.rb [hunk 3]
```diff
@@ -0,0 +1,668 @@
...
+        when 'Stock'
+          "#{day} Stock"
+        when 'Operating'
+          "#{day} Operating Round (#{round_number} of #{@operating_rounds})"
+        end
+      end
+
+      # Override this, and add elements (paragraphs of text) here to display it on Info page.
+      def timeline
+        @timeline.append("Game ends after OR #{@last_or}")
```
Comment (tobymao) [inline lib/engine/game/g_18_zoo.rb:645]: @timeline <<

File: lib/engine/game/g_18_zoo.rb [hunk 4]
```diff
@@ -0,0 +1,668 @@
...
+        @timeline.append("Next set of Operating Rounds will have #{OR_SETS[@turn - 1]} ORs")
+      end
+
+      def add_cousins
+        @log << 'Cousins join families.'
+
+        reserve_shares(true)
+      end
+
+      def reserve_shares(purchasable)
```
Comment (tobymao) [inline lib/engine/game/g_18_zoo.rb:655]: why is this called reserve share if you can pass true or false to it? should it be named something else?

File: lib/engine/game/g_18_zoo.rb [hunk 5]
```diff
@@ -0,0 +1,668 @@
...
+        @corporations.each { |c| c.shares.last.buyable = purchasable }
+      end
+
+      def update_current_and_future(to_current, to_future, turn)
+        @available_companies.concat(to_current) if @turn == turn
+        return unless !@all_private_visible && to_future && @turn == turn
```
Comment (tobymao) [inline lib/engine/game/g_18_zoo.rb:661]: don't use unless with multiple statements, change it to the equivalent if statement

File: lib/engine/game/g_18_zoo.rb (hunk 1 above)
Comment (carlorusso) [reply lib/engine/game/g_18_zoo.rb]: done in 558751e5

File: lib/engine/game/g_18_zoo.rb (hunk 2 above)
Comment (carlorusso) [reply lib/engine/game/g_18_zoo.rb]: done in 558751e5

File: lib/engine/game/g_18_zoo.rb (hunk 3 above)
Comment (carlorusso) [reply lib/engine/game/g_18_zoo.rb]: I removed it, the text was not correct anyway.

File: lib/engine/game/g_18_zoo.rb (hunk 4 above)
Comment (carlorusso) [reply lib/engine/game/g_18_zoo.rb]: removed in 558751e5

File: lib/engine/game/g_18_zoo.rb (hunk 5 above)
Comment (carlorusso) [reply lib/engine/game/g_18_zoo.rb]: done in 558751e5

**Pull request merged**